File: Java/NetBeans2/QLNV/src/qlnv/DSNV.java
package qlnv;

import java.io.*;
import java.util.Vector;

public class DSNV {
    private Vector<NV> ds;

    public DSNV() {
        ds = new Vector<>();
    }
    
    public void them(NV a) {
        if (!ds.contains(a)) {
            ds.add(a);
        }
    }
    
    public void hienThi(String loaiNV) {
        int soLuongNV = ds.size();
        
        for (int i = 0; i < soLuongNV; i++) {
            NV nv = ds.elementAt(i);
            
            if (nv.loaiNV().equals(loaiNV)) {
                System.out.println(nv);
            }
        }
    }
    
    public float luongTB() {
        int soLuongNV = ds.size();
        double tongLuong = 0;
        
        for (int i = 0; i < soLuongNV; i++) {
            tongLuong = ds.elementAt(i).tinhLuong();
        }
        
        return (float) tongLuong / soLuongNV;
    }
    
    public void ghiTep() {
        try {
            FileOutputStream fos = new FileOutputStream("DSNV.dat");
            ObjectOutputStream ous = new ObjectOutputStream(fos);
            int soLuongNV = ds.size();
            
            for (int i = 0; i < soLuongNV; i++) {
                ous.writeObject(ds.elementAt(i));
            }
            
            fos.close();
            ous.close();
        } catch (IOException e) {
            System.out.println("Lỗi ghi tệp!");
        }
    }

    
}

File: C#/giuaKY/giuaKY/DSNV.cs
using System.Collections;

namespace giuaKY
{
    public class DSNV
    {
        private ArrayList ds;

        public DSNV()
        {
            ds = new ArrayList();
        }

        public void them(NV nv)
        {
            ds.Add(nv);
        }

        public void hienThi(string loaiNV)
        {
            foreach (NV item in ds)
            {
                if (item.loaiNV().Equals(loaiNV))
                {
                    Console.WriteLine(item);
                }
            }
        }

        public float luongTB()
        {
            float tongLuong = 0;

            foreach (NV item in ds)
            {
                tongLuong += item.tinhLuong();
            }

            return tongLuong / ds.Count;
        }
    }
}
